### PR TITLE
RN-44: Adds flag SVG to post header.

### PR DESCRIPTION
**Pull request by mkraft**
#### Summary
Adds SVG flag to post headers.

#### Ticket Link
[RN-44](https://mattermost.atlassian.net/browse/RN-44)

#### Checklist
- [ ] Added or updated unit tests (required for all new features)
- [x] Has UI changes

#### Device Information
This PR was tested on:
* iPhone X, iOS 11 (emulated)
* iPhone 6, iOS 11 (emulated)
* iPhone 6s, iOS 11
* Nexus 6P, Android 8.0 (emulated)
* Nexus 5, Android 6

#### Screenshots
screenshot_1507384629
simulator screen shot - iphone x - 2017-10-07 at 09 57 21

### CODE REVIEW

File: app/components/post_header/index.js [hunk 1]
```diff
@@ -36,7 +37,8 @@ function makeMapStateToProps() {
             isPendingOrFailedPost: isPostPendingOrFailed(post),
             isSystemMessage: isSystemMessage(post),
             overrideUsername: post.props && post.props.override_username,
-            theme: getTheme(state)
+            theme: getTheme(state),
+            isFlagged: isPostFlagged(post.id, myPreferences)
```
Comment (enahum) [inline app/components/post_header/index.js]: I wonder if we should get this in `post` and pass it as an ownProp to post_header and post_body as with this change we are currently looking in both places if the post is flagged

File: app/components/post_header/post_header.js [hunk 2]
```diff
@@ -241,7 +253,8 @@ const getStyleSheet = makeStyleSheetFromTheme((theme) => {
             fontSize: 13,
             marginLeft: 5,
             marginBottom: 1,
-            opacity: 0.5
+            opacity: 0.5,
+            marginRight: 5
```
Comment (enahum) [inline app/components/post_header/post_header.js]: maybe you don't need this if you add the margin to the flagContainer style below

File: app/components/post_header/index.js (hunk 1 above)
Comment (mkraft) [reply app/components/post_header/index.js]: @enahum Yes that makes sense. I'll make the change.

File: app/components/post_header/post_header.js (hunk 2 above)
Comment (mkraft) [reply app/components/post_header/post_header.js]: @enahum I was mimicking what was done with `displayName` and `time` where their combined 10dp spacing came from 5dp each. I could put the whole 10dp on the `flagContainer` though. Thoughts?

Comment (enahum) [reply app/components/post_header/post_header.js]: yes I think is best to do it in the `flagContainer` the other two are always shown and the *flag* is only shown if the post has been flagged

Comment (mkraft) [reply app/components/post_header/post_header.js]: @enahum Cool, I'll put the 10dp on the `flagContainer`.

**Comment by enahum** [comment]
Hi @mkraft 

apart from the changes requested by @hmhealey this PR will need a rebase

**Comment by enahum** [comment]
@mkraft you have some build failures and try and rebase instead of merging ;)

**Comment by mkraft** [comment]
@enahum Yes I can see the build errors and will fix them. I can rebase instead in future to keep history clean - in another PR I thought someone said it didn't matter. Regarding...
> Clicking on the flag should unflag the post according to the ticket
There were discussions about this last month when I did the story and my understanding was that clicking the flag wasn't consistent with the UX (that instead we want to continue using the long press, which makes sense to me). I'll realign the flag for now.

**Comment by hmhealey** [comment]
Sounds good, but the ticket should've been updated for that

Also, I don't think we have any real preference between merging and rebasing to update PRs. GitHub squashes and rebases all the changes anyway, so it shouldn't matter how you keep the branch up to date

**Comment by mkraft** [comment]
simulator screen shot - iphone 6 - 2017-11-09 at 17 34 14
screenshot_1510266882

**Comment by enahum** [comment]
@mkraft sorry dude but I think is still not aligned, what do you think?

image

**Comment by mkraft** [comment]
@enahum I liked the look of it better before when it was raised higher.

**Comment by enahum** [comment]
@lfbrock thoughts on this?

**Comment by asaadmahmood** [comment]
Looks like that screen is a bit outdated.
Would prefer if we can use the same icon as the webapp, and that is a bit bigger than the current icon. The svg for this flag icon active/inactive both can be found in the webapp.

The size of the icon would be 20px by 20px.
Both icons can be found in this file:
https://github.com/mattermost/mattermost-webapp/blob/8d87153e4efed695394e5afad11c88506c2708d2/utils/constants.jsx#L522

32633128-476b8c48-c574-11e7-939a-26f21a4158cc

**Comment by mkraft** [comment]
@asaadmahmood Sounds good I'll update it.

**Comment by mkraft** [comment]
@asaadmahmood Here are screenshots using the filled SVG from the webapp with 10dip margin left and and width and height of 10dip. How does it look; did you want me to make any adjustments? Thanks.

simulator screen shot - iphone 6 - 2017-11-10 at 08 45 36
screenshot_1510321578

**Comment by asaadmahmood** [comment]
@mkraft Thanks, this look very close, I think maybe just increase it by 1px, and see how it looks?

**Comment by mkraft** [comment]
@asaadmahmood Here it is 1 dip larger; how does it look?

simulator screen shot - iphone 6 - 2017-11-13 at 09 27 14
screenshot_1510583237

**Comment by asaadmahmood** [comment]
@mkraft Can we move the icon 1px up for the second screenshot? I think we would ideally want the bottom to be aligned (or close to aligned) with the text.

**Comment by mkraft** [comment]
@asaadmahmood Did you want it 1 up for both Android and iOS or just Android?

**Comment by asaadmahmood** [comment]
2px for IOS, 1px for Android. @mkraft

**Comment by mkraft** [comment]
@asaadmahmood How's this?

simulator screen shot - iphone 6 - 2017-11-13 at 16 35 15
screenshot_1510608919

**Pull request merged**